src/axom/inlet/tests/inlet_Inlet.cpp

```diff
@@ -1236,6 +1235,100 @@ TEST(inlet_verify, verifyTableLambda3) {
+  LuaReader lr;
+  lr.parseString(testString);
+
+  std::unordered_map<int,int> ints;
+  bool found = lr.getIntMap("luaArray", ints);
+  EXPECT_TRUE(found);
+  std::unordered_map<int,int> expectedInts{{1,4},{2,5},{3,6},{12,2}};
+  EXPECT_EQ(expectedInts, ints);
+
+  std::unordered_map<int, double> doubles;
+  found = lr.getDoubleMap("luaArray", doubles);
+  EXPECT_TRUE(found);
+  std::unordered_map<int,double> expectedDoubles{{1,4},{2,5},{3,6},{12,2.4}};
+  EXPECT_EQ(expectedDoubles, doubles);
+
+  std::unordered_map<int, bool> bools;
+  found = lr.getBoolMap("luaArray", bools);
+  EXPECT_TRUE(found);
+  std::unordered_map<int, bool> expectedBools{{4, true}, {8,false}};
+  EXPECT_EQ(expectedBools, bools);
+
+  std::unordered_map<int, std::string> strs;
+  found = lr.getStringMap("luaArray", strs);
+  EXPECT_TRUE(found);
+  std::unordered_map<int, std::string> expectedStrs{{33, "hello"}, {200, "bye"}};
+  EXPECT_EQ(expectedStrs, strs);
+}
+
+TEST(lua_Arrays, intInletArrays) {
+  DataStore ds;
+  std::string testString = "luaArrays = { arr1 = { [1] = 4, [2] = 5, [3] = 6 , [12] = 2.4}, arr2 = {[4] = true, [8] = false}, arr3 = {[33] = 'hello', [2] = 'bye'}, arr4 = { [12] = 2.4 } }";
+  auto inlet = createBasicInlet(&ds, testString);
+
+  inlet->getGlobalTable()->addIntArray("luaArrays/arr1");
+
+  auto group = inlet->getGlobalTable()->sidreGroup()->getGroup("luaArrays/arr1/_inlet_array");
+  auto idx = group->getGroup("1");
+  EXPECT_TRUE(idx);
+  int val = idx->getView("value")->getScalar();
+  EXPECT_EQ(val, 4);
+
+  idx = group->getGroup("12");
+  EXPECT_TRUE(idx);
+  val = idx->getView("value")->getScalar();
+  EXPECT_EQ(val, 2);
+
+  idx = group->getGroup("2");
+  EXPECT_TRUE(idx);
+  val = idx->getView("value")->getScalar();
+  EXPECT_EQ(val, 5);
+
+  idx = group->getGroup("3");
+  EXPECT_TRUE(idx);
+  val = idx->getView("value")->getScalar();
+  EXPECT_EQ(val, 6);
+
+  inlet->getGlobalTable()->addBoolArray("luaArrays/arr2");
+  group = inlet->getTable("luaArrays/arr2/_inlet_array")->sidreGroup();
+
+  idx = group->getGroup("4");
+  EXPECT_TRUE(idx);
+  int8_t boolVal = idx->getView("value")->getScalar();
+  EXPECT_EQ(boolVal, 1);
+
+  idx = group->getGroup("8");
+  EXPECT_TRUE(idx);
+  boolVal = idx->getView("value")->getScalar();
+  EXPECT_EQ(boolVal, 0);
+
+  inlet->getGlobalTable()->addStringArray("luaArrays/arr3");
+  group = inlet->getTable("luaArrays/arr3/_inlet_array")->sidreGroup();
+
+  idx = group->getGroup("33");
+  EXPECT_TRUE(idx);
+  std::string str = idx->getView("value")->getString();
+  EXPECT_EQ(str, "hello");
+
+  idx = group->getGroup("2");
+  EXPECT_TRUE(idx);
+  str = idx->getView("value")->getString();
+  EXPECT_EQ(str, "bye");
+
+  inlet->getGlobalTable()->addDoubleArray("luaArrays/arr4");
+  group = inlet->getTable("luaArrays/arr4/_inlet_array")->sidreGroup();
+
+  idx = group->getGroup("12");
+  EXPECT_TRUE(idx);
+  double doubleVal = idx->getView("value")->getScalar();
+  EXPECT_EQ(doubleVal, 2.4);
+
+}
+
```

Comment:
There needs to be more tests of each function added.  There are several new functions that are not called at all in any tests.  Also an example of how the users would use the new functionality is needed.

Reply:
Agreed on the example. As for the tests, all of the new functions seem to be included in the tests. Could you please point out which function(s) I missed?

Reply:
Oh i see.. you are doing the addArray's in this test.. then the getMaps above...